src/docs/arch-tiered-storage.dox

```diff
@@ -0,0 +1,439 @@
+this aligns with the \c "tier:A" entry in the metadata.
+
+Also in the \c WT_TIERED struct is information about the current
+object id, and how many object ids are known. If we know that the
+current object id is 4, we can construct the needed name: \c
+"file:A-0000000004.wtobj", likewise, previous cloud object names can
+be constructed, like \c "object:A-0000000003.wtobj".
+Navigation like this, constructing names to look up in metadata or
+as data handles doesn't happen often, mostly during startup or
+flush_tier. During high performance paths, the \c WT_TIERED struct
+or the \c WT_BM (block manager struct) associated with the Btree have
+everything we need.
+
+A "tiered:" entry (associated with a tiered table) and a "file:" entry
+(associated with a non-tiered table) behave almost identically in
+the WiredTiger system.  In fact, the \c WT_BTREE_PREFIX macro checks
+to see if a URI matches either one of these prefix strings.  The macro basically
+means "does this thing walk and talk like a Btree?".  In both cases,
+the dhandle found with the given name has a \c dhandle->handle that
+points to the open \c WT_BTREE.
+
+The \c "object:" entries do not have separate in-memory data structures
+(there is a \c WT_TIERED_OBJECT defined in \c tiered.h, but it is not used).
+And while \c "table:", \c "file:" and "tiered:" all have data handles
+using those URIs, there are no separate data handles for each object.
+This is probably a good thing, as scaling the number of data handles
+system-wide has been challenging.
+
+The following table summarizes the various relationships.
+
+URI prefix    | struct type         | has dhandle?  |  dhandle notes
+------------- | ------------------- | ------------- | --------------------------------------
+\c table:     | WT_TABLE            | yes           | the dhandle is cast to \c (WT_TABLE *)
+\c file:      | (none)              | yes, but...   | does not relate to a Btree
+\c tiered:    | WT_TIERED, WT_BTREE | yes           | the dhandle is cast to \c (WT_TIERED *), and dhandle->handle is a \c (WT_BTREE *)
+\c tier:      | WT_TIERED_TIERS     | no            | WT_TIERED->tiers[1] is a (WT_TIERED_TIERS *)
+\c object:    | (none)              | no            | no dhandle for objects
+
+@section ts_buckets Storage Sources, Buckets, and Prefixes
+
+The location of objects in the cloud requires some information to be
+stored: the cloud provider, the bucket used, and an id to reference
+any needed credentials in a key management system. When a connection
+is opened for tiered storage, these are specified and are used in
+the default case.  However, we want the ability to have some tables stored in
+different places, so we need cloud location information to live
+in the \c WT_TIERED object as well. To make this happen, all this
+\a location information is abstracted into a \c WT_BUCKET_STORAGE .
+The connection has a pointer to a \c WT_BUCKET_STORAGE, and each
+\c WT_TIERED does as well.
```

Comment:
Given my comment earlier about being enabled, perhaps this section should move to the top.

Reply:
I added some information about enabled to the introduction, but I think I'd like to keep this here, as it seems too detailed for the introduction.